pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -587,66 +650,54 @@ func (r *MachineSetSyncReconciler) applySynchronizedConditionWithPatch(ctx conte
 }
 
 // createOrUpdateCAPIInfraMachineTemplate creates a CAPI infra machine template from a MAPI machine set, or updates if it exists and it is out of date.
-func (r *MachineSetSyncReconciler) createOrUpdateCAPIInfraMachineTemplate(ctx context.Context, mapiMachineSet *machinev1beta1.MachineSet, infraMachineTemplate client.Object, newCAPIInfraMachineTemplate client.Object) (ctrl.Result, error) { //nolint:unparam
+func (r *MachineSetSyncReconciler) createOrUpdateCAPIInfraMachineTemplate(ctx context.Context, mapiMachineSet *machinev1beta1.MachineSet, infraMachineTemplate client.Object, newCAPIInfraMachineTemplate client.Object) error {
```

Comment:
When we are updating. Is there a path that allows the objectmeta (labels/annotations) to be updated without deleting/recreating the InfraMachineTemplate?

Reason I ask, is that the objectmeta is not included in the spec hash, so in theory we could generate two identically named templates, when the only difference is is objectmeta, and I think this would break if we aren't handling this case specifically as an update to the existing resource

Reply:
I think this is now fixed with use of Patch. If only labels or annotations differ, they will be updated in place.